tests/providers/integration_tests.py

```diff
@@ -95,6 +96,8 @@ def __init__(self):
         self.provider_module = None
 
     def setup_method(self, _):
+        warnings.filterwarnings("ignore", category=ResourceWarning,
```

Comment:
I am concerned by this filter. The test is generating alerts about unclosed sockets? Seems like a problem of resource cleaning in the provider implementation.